api/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -107,9 +107,11 @@ type OpenStackDataPlaneServiceSpec struct {
 	// github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1
 	ContainerImageFields []string `json:"containerImageFields,omitempty" yaml:"containerImageFields,omitempty"`
 
-	// EDPMServiceName - name to use for edpm_service_name ansible variable
-	// +kubebuilder:validation:Optional
-	EDPMServiceName string `json:"edpmServiceName,omitempty" yaml:"edpmServiceName,omitempty"`
+	// EDPMServiceType - service type, which typically corresponds to one of
+	// the default service names (such as nova, ovn, etc). Also typically
+	// corresponds to the ansible role name (without the "edpm_" prefix) used
+	// to manage the service.
+	EDPMServiceType string `json:"edpmServiceType,omitempty" yaml:"edpmServiceType,omitempty"`
```

Comment:
As far I see all the usage of EDPMServiceType considers the service.Name as a default value if no explicit value is set. If so then I think we can simplify the implementation by doing such defaulting logic in the defaulting webhook. That would allow us to remove a couple of condition in the controller implementation.

Reply:
I've switched it to use a webhook

Reply:
so now that we have this field can we update the EDPMServiceType for all of the standard dataplane services.

i.e. set it to nova in https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml

that will enable this to work transparently for install yamls since it is just modifying the default one to generate the nova-custom one but for ci frame work since they start form scratch wee will still need to update there definition.

Reply:
we could also do that in a followup if we want to keep this small and move forward with it as is

Reply:
thanks for the webhook change, looks good.